### PR TITLE
[DEV-5222] Make repositories optional in ForSetPullRequests

### PR DESCRIPTION
**Pull request by gaetano-guerriero**

### CODE REVIEW

File: openapi.yaml [hunk 1]
```diff
@@ -7856,9 +7854,14 @@ components:
         account: 1
       properties:
         for:
-          description: List of repository groups for which to calculate the metrics.
+          description: |
+            List of repository groups for which to calculate the metrics.
+            Use `null` as repository group to refer to every repository.
```
Comment (vmarkovtsev) [inline openapi.yaml:7859]:
```suggestion
            Use `null` as repository group to match every repository.
```

Comment (vmarkovtsev) [reply openapi.yaml:7859]: + we should mention that an empty list is going to match zero repositories.

File: openapi.yaml [hunk 2]
```diff
@@ -4408,8 +4408,6 @@ components:
           maxItems: 64
           minItems: 1
           type: array
-      required:
```
Comment (vmarkovtsev) [inline openapi.yaml:4411]: Let's include extra docs about the meaning of empty and missing/null `repositories`, similar to releases.

File: openapi.yaml [hunk 3]
```diff
@@ -7856,9 +7854,14 @@ components:
         account: 1
       properties:
         for:
-          description: List of repository groups for which to calculate the metrics.
+          description: |
+            List of repository groups for which to calculate the metrics.
+            Use `null` as repository group to refer to every repository.
           items:
-            $ref: '#/components/schemas/RepositorySet'
+            items:
```
Comment (vmarkovtsev) [inline openapi.yaml:7861]: We should leave the reference to RepositorySet and use my hack on adding a description that I detailed in private-api-spec

**Review by vmarkovtsev — CHANGES_REQUESTED** [review]

File: openapi.yaml (hunk 2 above)
Comment (gaetano-guerriero) [reply openapi.yaml:4411]: added to `repositories` field with allOf/$ref-description trick

File: openapi.yaml (hunk 3 above)
Comment (gaetano-guerriero) [reply openapi.yaml]: I put RepositorySet reference back, but in the second element of anyOf I need anyway to replicate RepositoryID

File: openapi.yaml [hunk 4]
```diff
@@ -7856,9 +7858,17 @@ components:
...
+            List of repository groups for which to calculate the metrics.
+            An empty repository (empty array) group will match nothing.
+            Use `null` as repository group to match every repository.
+          items:
+            anyOf:
+            - $ref: '#/components/schemas/RepositorySet'
+            - items:
+                $ref: '#/components/schemas/RepositoryID'
+              nullable: true
+              type: array
```
Comment (vmarkovtsev) [inline openapi.yaml:7871]: This should work better
```suggestion
            oneOf:
            - $ref: '#/components/schemas/RepositorySet'
            - type: "null"
```

**Review by vmarkovtsev — CHANGES_REQUESTED** [review]

File: openapi.yaml (hunk 4 above)
Comment (gaetano-guerriero) [reply openapi.yaml]: type: 'null' is not supported by openapi even if supported by jsonschema
https://swagger.io/docs/specification/data-models/data-types/
luckily this is working, it's simple enough:
```
            oneOf:
            - $ref: '#/components/schemas/RepositorySet'
            - enum:
              - null
```

**Review by vmarkovtsev — APPROVED** [review]

**Review by baristna — APPROVED** [review]

**Pull request merged**